### PR TITLE
Challenge-07-Added functionality to the likes buttons

### PR DESCRIPTION
**Pull request by messat**
1. Added 3 buttons with functionality to optimistically render the page.

### CODE REVIEW

**Comment by messat** [comment]
Merging branch from Challenge 07 to main

File: src/components/eventListener/Votes.jsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+
+
+
+
+
+
+
+
+// function UpvotesClick (article_id){
```
Comment (nc-haz-mallion) [inline src/components/eventListener/Votes.jsx:88]: See prev comment about commented out code

File: src/utils/api.js [hunk 2]
```diff
@@ -6,15 +6,44 @@ const instance = axios.create({
 
 function getAllArticles (){
   return instance.get("/articles")
+  .then(({data})=>{
+    const articlesArr = data.articles.sort((a, b)=>{
```
Comment (nc-haz-mallion) [inline src/utils/api.js:10]: does the api not sort the articles for you? 
I' not sure you need them to be arranged by ID, but if you do this should be extracted into its own (fully tested) util function.

File: src/utils/api.js [hunk 3]
```diff
@@ -6,15 +6,44 @@ const instance = axios.create({
...
+        return  a.article_id -b.article_id
+     })
+     return articlesArr
+  })
 }
 
 function getSingleArticle (article_id){
     return instance.get(`/articles/${article_id}`)
+    .then(({data})=>{
```
Comment (nc-haz-mallion) [inline src/utils/api.js:19]: it might be neater to just return the `data.article`

**Review by nc-haz-mallion — COMMENTED** [review]
Some really nice code here. 
Please read through the comments. 

In the future please attempt to be more descriptive with your commit messages - there is a lot of code here and only one commit.

**Pull request merged**